core/src/main/java/org/infinispan/remoting/inboundhandler/GlobalInboundInvocationHandler.java

```diff
@@ -153,7 +154,7 @@ private void runReplicableCommand(ReplicableCommand command, Reply reply, boolea
       try {
          invokeReplicableCommand(command, reply, preserveOrder);
       } catch (Throwable throwable) {
-         if (throwable.getCause() != null && throwable instanceof CompletionException) {
+         if (throwable.getCause() != null && (throwable instanceof CompletionException || throwable instanceof CacheConfigurationException)) {
```

Comment:
I agree that seems much cleaner to just catch and rethrow IllegalLifecycleStateException.  However this doesn't prevent if we got an InterruptedException wrapped by a CacheConfigurationException.  

So I would say either to just have this change or to have both.